File: Java/program1/Main.java
import java.util.*;

public class Main {
       
    // Функция будет возвращать объект Object[], содержащий количество операций и список строк
    public static Object[] task1(String s) {
        int n = s.length();
        List<String> steps = new ArrayList<>();
        StringBuilder current = new StringBuilder();
        
        for (int i = n / 2; i > 0; i--) {
            if (s.substring(0, i).equals(s.substring(i, 2*i))) {
                for (int j = 0; j < i; j++) {
                    current.append(s.charAt(j));
                    steps.add(current.toString());
                }
                current.append(current);
                steps.add(current.toString());
                current = new StringBuilder();
                for (int j = 2*i; j < n; j++) {
                    current.append(s.charAt(j));
                    steps.add(current.toString());
                }
                return new Object[]{i + 1 + (n - 2*i), steps};
            }
        }
        
        for (int i = 0; i < n; i++) {
            current.append(s.charAt(i));
            steps.add(current.toString());
        }
        return new Object[]{n, steps};
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine().trim();
        Object[] result = task1(input);

	int count = (int)result[0];
	@SuppressWarnings("unchecked")
	List<String> steps = (List<String>)result[1];
        
    System.out.print(result[0] + " = ");
	System.out.println(String.join("->", steps));
    }
}

File: Rust/program1/src/main.rs
use std::io;

fn task1(s: &str) -> (usize, Vec<String>) {
    let n = s.len();
    let mut steps = vec![];
    let mut current = String::new();
    
    for i in (1..=n/2).rev() {
        if s[0..i] == s[i..2*i] {
            for j in 0..i {
                current.push(s.chars().nth(j).unwrap());
                steps.push(current.clone());
            }
            let temp = current.clone();  // Исправление здесь
            current.push_str(&temp);
            steps.push(current.clone());
            current.clear();
            for j in 2*i..n {
                current.push(s.chars().nth(j).unwrap());
                steps.push(current.clone());
            }
            return (i + 1 + (n - 2*i), steps);
        }
    }
    
    for c in s.chars() {
        current.push(c);
        steps.push(current.clone());
    }
    (n, steps)
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let input = input.trim();
    
    let (count, steps) = task1(input);
    
    print!("{} = ", count);
    for (i, step) in steps.iter().enumerate() {
        print!("{}", step);
        if i != steps.len() - 1 {
            print!("->");
        }
    }
    print!("\n");
}

